jans-cedarling/cedarling/src/models/token_data.rs

```diff
@@ -23,6 +23,14 @@ impl TokenPayload {
         Self { payload }
     }
 
+    /// Clone current value and extend with the given payload.
+    pub fn extend(&self, payload: TokenPayload) -> Self {
+        let mut result = self.clone();
+        result.payload.extend(payload.payload);
+
+        result
+    }
```

Comment:
To avoid the cloning 
```suggestion
     pub fn extend(&mut self, payload: TokenPayload) {
        self.payload.extend(payload.payload);
    }
```

Reply:
If I do this, we also need clone token payload, but in another place.
Or if I use mutable reference everywhere, it can lead to bugs in the future.


Also, TokenPayload it is `Value object` it is not `Entity`. 

We can't just get payload of one token and fill it with data of other token. 
In the result, it will be new value.

So I don't want to make this method take mutable reference. Maybe we should rename it...